elastic/kibana
# [Stack Monitoring] fix ccr read_exceptions alert

## Summary

Closes https://github.com/elastic/kibana/issues/153298

## Testing
- Setup CCR ([see how to](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/dev_docs/how_to/running_components_from_source.md#multi-cluster-tests-for-ccrccs-or-listing). I used two cloud clusters, happy to provide credentials to reviewers to avoid the setup)
  - Setup CCR between two clusters, create a follower indice and replicate some data
  - Intentionally break the remote cluster connection on the follower cluster (update the cluster endpoint). At this point read_exceptions should appear in ccr documents
- Start local stack
  - local elasticsearch should have [this change](https://github.com/elastic/elasticsearch/pull/94875). See [howto run component from source](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/dev_docs/how_to/running_components_from_source.md#single-cluster-testing)
- Enable stack monitoring default rules
- Start metricbeat collection of the follower cluster
  - Ensure ccr alert triggers with metricbeat 7.x (this won't work against cloud cluster because the license is not supported in 7.x but support was added later, I'll see if we can [backport this change](https://github.com/elastic/beats/pull/34105))
  - Ensure ccr alert triggers with metricbeat 8.x (build should include [this change](https://github.com/elastic/beats/pull/34957))
